mtags-java/src/main/scala/scala/meta/internal/pc/JavaCompletionProvider.scala

```diff
@@ -52,32 +53,27 @@ class JavaCompletionProvider(
     node match {
       case Some(n) =>
         val items = n.getLeaf.getKind match {
-          case MEMBER_SELECT => completeMemberSelect(task, n)
-          case IDENTIFIER => completeIdentifier(task, n) ++ keywords(n)
+          case MEMBER_SELECT => completeMemberSelect(task, n).distinct
+          case IDENTIFIER => completeIdentifier(task, n).distinct ++ keywords(n)
           case _ => keywords(n)
         }
-        new CompletionList(items.distinct.sorted(ordering).asJava)
+        new CompletionList(items.asJava)
       case None => new CompletionList()
     }
   }
 
-  private def ordering: Ordering[CompletionItem] = {
-    val identifier = extractIdentifier.toLowerCase
-
-    new Ordering[CompletionItem] {
-      def score(item: CompletionItem): Int = {
-        val name = item.getLabel.toLowerCase
-
-        if (name.startsWith(identifier)) 0
-        else if (name.contains(identifier)) 1
-        else 2
-      }
-
-      override def compare(i1: CompletionItem, i2: CompletionItem): Int = {
-        java.lang.Integer.compare(score(i1), score(i2))
-      }
-    }
+  private def identifierScore(element: Element): Int = {
+    val name = element.getSimpleName().toString().toLowerCase()
+    if (name.startsWith(identifier)) 0
+    else if (name.contains(identifier)) 1
+    else 2
```

Comment:
Optional optimization
```suggestion
   name.indexOf(identifier) match {
     case 0 => 0
     case -1 => 2
     case _ => 1
   }
```